lib/common/client/configuration.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require 'singleton'
+
+module Common
+  module Client
+    class Configuration
+      include Singleton
```

Comment:
Is there any reason to make this a Singleton as opposed to just overriding a handful of class methods in subclasses?